File: server/kafka-backend/src/infrastructure/adapters/in/rest/login.controller.ts
import
{
    Controller,
    Post,
    Body,
    Logger,
    HttpCode,
    HttpStatus,
} from '@nestjs/common';
import { InjectDataSource } from '@nestjs/typeorm';
import { DataSource } from 'typeorm';
import { UserOrmEntity } from '../../out/persistence/typeorm/entities/user.orm-entity';

@Controller('login')
export class LoginController
{
    private readonly logger = new Logger(LoginController.name);

    constructor(
        @InjectDataSource()
        private readonly dataSource: DataSource,
    ) {}

    @Post()
    @HttpCode(HttpStatus.OK)
    async login(@Body() body: { phone?: string })
    {
        const { phone } = body;

        if(!phone || typeof phone !== 'string')
        {
            return { statusCode: 400, error: 'Numar de telefon invalid' };
        }

        try
        {
            const userRepo = this.dataSource.getRepository(UserOrmEntity);
            const user = await userRepo.findOne(
            {
                where: { nrTelefon: phone },
                select: { id: true, contAprobat: true, termeniAcceptati: true },
            });

            if(!user)
            {
                return { exists: false };
            }

            return {
                exists: true,
                userId: user.id,
                approved: user.contAprobat,
                acceptedterms: user.termeniAcceptati,
            };
        }
        catch (err)
        {
            this.logger.error(err, 'Eroare la baza de date (login)');
            return { statusCode: 500, error: 'Eroare la comunicarea cu serverul' };
        }
    }
}

File: server/backend/src/main/java/com/intbank/infrastructure/rest/LoginController.java
package com.intbank.infrastructure.rest;

import com.intbank.infrastructure.persistence.repository.UserJpaRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/login")
public class LoginController
{

    private static final Logger log = LoggerFactory.getLogger(LoginController.class);
    private final UserJpaRepository userRepo;

    public LoginController(UserJpaRepository userRepo)
    {
        this.userRepo = userRepo;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.OK)
    public Map<String, Object> login(@RequestBody Map<String, String> body)
    {
        String phone = body.get("phone");
        if (phone == null || phone.isBlank()) {
            return Map.of("statusCode", 400, "error", "Numar de telefon invalid");
        }

        try {
            var user = userRepo.findByNrTelefon(phone);
            if (user.isEmpty()) {
                return Map.of("exists", false);
            }
            var u = user.get();
            return Map.of(
                    "exists", true,
                    "userId", u.getId(),
                    "approved", u.getContAprobat() != null && u.getContAprobat(),
                    "acceptedterms", u.getTermeniAcceptati() != null && u.getTermeniAcceptati()
            );
        } catch (Exception err) {
            log.error("Eroare la baza de date (login)", err);
            return Map.of("statusCode", 500, "error", "Eroare la comunicarea cu serverul");
        }
    }
}